operator/docs/enhancements/zone_aware_replication.md

```diff
@@ -0,0 +1,358 @@
+     if err != nil {...}
+     if pod.Annotations == nil {
+        pod.Annotations = make(map[string]string)
+     }
+     // 2. Add topology keys to the Pod's annotations
+     for k, v := range kv {
+        if k == topologyKey {
+          pod.Annotations[k] = v
+        }
+     }
+     // 3. Update the Pod
+     return wh.client.Update(ctx, pod)
+  }); err != nil {...}
+}
+```
+Some reference impementations on how to use the a `pods/binding` sub-resource mutating webhook can be found here:
+* [elastic/cloud-on-k8s](https://github.com/elastic/cloud-on-k8s/issues/3933#issuecomment-950581345)
+* [kubemod/kubemod](https://github.com/kubemod/kubemod/commit/95c43bc7d532db05051a83df5ad97e36a0f5d7e2#diff-d63889fb133b18cb690d365c25f43bf0580f7a7aaa91ffa03b121b9b6b5f2d60)
+
+2. The second solution is to use the operator to modify the pods once they are scheduled. The operator checks if the `pod.spec.topologySpreadConstraints.topologyKey` is set. If yes, then the operator extracts the annotation(key and value) from the node, and sets it as an annotation for the pod by patching the pod. The zone value can then be accessed from within the container as an environment variable just like in solution 1 by adding the env variable in the container. For reference consider:
+* [elastic/cloud-on-k8s implementation](https://github.com/elastic/cloud-on-k8s/blob/85aa75b7a7765330475c0b92a0851939ac3d9578/pkg/controller/elasticsearch/driver/node_labels.go#L57) 
+* [elastic/cloud-on-k8s discussion](https://github.com/elastic/cloud-on-k8s/issues/3933#issuecomment-962897004)
+
+This is how the expected individual pod spec will look like after the webhook has added the topology key annotation in the pod:
+
+```yaml
+kind: Pod
+apiVersion: v1
+metadata:
+  name: ingester
+  annotations: 
+    topology.kubernetes.io/zone: zone-a 
+spec:
+  topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: topology.kubernetes.io/zone
+    whenUnsatisfiable: DoNotSchedule
+    labelSelector:
+      matchLabels:
+        ingester: pod
+  containers:
+  - name: ingester
+    env:
+    - name: ZONE
+        valueFrom:
+          fieldRef:
+            fieldPath: metadata.annotations['topology.kubernetes.io/zone']
+```
+
+The container can then be started by passing the configuration via the cli flag `instance_availability_zone: $ZONE`
+
+
+### Open Questions
+
+* Both these methods first modify the pod to add the annotation[topology.kubernetes.io/zone: zone-a] in the pod and then modify the container to add a new env var or a volume which picks the zone value from the pod annotation via the downwardapi
+* In the Webhook method, is it safe to assume both the pod modification, followed by the container modification can be handled within the webhook?
+* In the operator method, there is going to be two merge patches of the object, one for the pod to add the annotation, and then the container to add the env variable.
```

Comment:
I don't see why the configuration of the environment variable needs to be added by the webhook (does not matter which of the variants). I don't think there's validation on whether the value provided by the `fieldRef` is set, which should mean that the environment variable configuration can be part of the staticly generated `podSpec`.